src/common/getType.ts

```diff
@@ -579,60 +717,17 @@ export function getCollectionTypes(expr: Expr, program: Program): ValueType[] {
   throw new PolygolfError("Type error. Node is not a collection.", expr.source);
 }
 
-/*
-function floorDiv(a: bigint, b: bigint): bigint {
-  const res = a / b;
-  return a < 0 !== b < 0 ? res - 1n : res;
+function getIntegerTypeMod(a: IntegerType, b: IntegerType): IntegerType {
+  const values: IntegerBound[] = [];
+  if (lt(b.low, 0n))
+    values.push(sub(1n, min(b.low, min(abs(a.low), abs(a.high)))));
+  if (lt(0n, b.high))
+    values.push(sub(min(b.low, min(abs(a.low), abs(a.high))), 1n));
```

Comment:
This doesn't look quite right, e.g. fails for `a={low:-2,high:2}` and `b={low:10,high:10}`. Currently pushes `1`, but should push `9` since -1 mod 10 gives 9. Might be better to just `values.push(sub(b.low, 1n));` since the extra tightness from knowing the interval of `a` doesn't matter much. Similar for the `b.low < 0` case.

Reply:
I fixed it using your suggestion and added a case for the constant inputs.